hibernate-core/src/main/java/org/hibernate/resource/jdbc/internal/ResourceRegistryStandardImpl.java

```diff
@@ -370,4 +386,21 @@ private boolean hasRegistered(final HashMap resource) {
 	private boolean hasRegistered(final ArrayList resource) {
 		return resource != null && !resource.isEmpty();
 	}
+	
+	private class StatementRegistryEntry {
```

Comment:
Seems the inner class should be static one for seemingly it doesn't need access to enclosing object.

Reply:
Good catch. It's totally better to be a nested one, especially for performance and its impact on GC.